.github/workflows/ci-cpp-build-gnoi.yml

```diff
@@ -0,0 +1,33 @@
+name: "bazel build"
+  
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  schedule:
+    - cron: "0 0 * * *"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      BAZEL: bazelisk-linux-amd64
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+    
+    - name: Mount bazel cache
+      uses: actions/cache@v2
+      with:
+        # See https://docs.bazel.build/versions/master/output_directories.html
+        path: "~/.cache/bazel"
+        # Create a new cache entry whenever Bazel files change.
+        # See https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows
+        key: bazel-${{ runner.os }}-build-${{ hashFiles('**/*.bzl', '**/*.bazel') }}
+        restore-keys: |
+          bazel-${{ runner.os }}-build-
+    - name: Install bazelisk
+      run: |
+        curl -LO "https://github.com/bazelbuild/bazelisk/releases/download/v1.8.1/$BAZEL"
```

Comment:
Apologies, I just realized that GitHub review comments get associated with an entire line, rather than specific characters one marks. So it looks like some of my previous comments we're confusing since they were out of context.

We removed a few thing too much here. We'll want to add back the following:
```
    - name: Install bazelisk
      run: |
        curl -LO "https://github.com/bazelbuild/bazelisk/releases/download/v1.8.1/$BAZEL"
        chmod +x $BAZEL
        sudo mv $BAZEL /usr/local/bin/bazel
        
      name: Build
      run: bazel build //...
```